lib/passport-wsfed-saml2/saml.js

```diff
@@ -97,6 +96,12 @@ SAML.prototype.validateSignature = function (xml, options, callback) {
     return callback(e);
   }
 
+  if (self.options.checkCertExpiration && !self.validateCertExpiration(xml)) {
+    valid = false;
+    return callback(new Error('SAML certificate not valid.'), null); //Note - could say "not currently valid" but
+                            // is it a risk to reveal that this is a date issue and not a generally invalid cert?
```

Comment:
If it is a risk to include that information in the response sent by the server, then we should at least put that information in a tenant log entry, as clearly as possible (preferably with start-end dates) so that customers can find what's wrong on their own.

Reply:
I think that's a solid direction. We'd need to ensure to attach some info to this error to bubble up to the service's error handlers.

Reply:
We should keep the external error message as is. I personally don't see any problems of mentioning the error being an expired certificate in the tenant log entry though. The cert start/end date may be unnecessary since the tenant should have access to the certificate and can check the expiration manually if need be.